code/game/objects/items/explosives/grenades/marines.dm

```diff
@@ -228,7 +232,7 @@
 	R.intensityfire = burn_level
 	R.rangefire = radius
 
-	new /obj/flamer_fire(T, cause_data, R, R.rangefire, null, flameshape, target)
+	new /obj/flamer_fire(T, cause_data, R, R.rangefire, null, flameshape, target, , , fire_type)
```

Comment:
just use fire_type = fire_type

Reply:
Doesn't work, likely since it's because it's recursive and using the same var between both.